src/etc/inc/filter.inc

```diff
@@ -2388,6 +2388,16 @@ function filter_generate_user_rule(&$FilterIflist, $rule)
     if (isset($rule['protocol']) && $rule['protocol'] == "icmp" && $rule['icmptype'] && $rule['ipprotocol'] == "inet6") {
         $aline['icmp6-type'] = "icmp6-type {$rule['icmptype']} ";
     }
+
+    if (!empty($rule['priority'])) {
+    	// need to use a hack for priority values since rule apparantly cannot store "0"
+		$aline['priority'] = ' set prio '.($rule['priority'] - 1).' ';
```

Comment:
I'm not sure what you mean, $rule['priority'] == "" won't store empty values, to be on the safe site for parameter initialisation, you could add isset($rule['priority'])) && $rule['priority'] == ""

Or do you mean something else?